File: services/cadastro-api/2-Application/Cadastro.Application/Dashboard/DashboardResumoResponse.cs
namespace Cadastro.Application.Dashboard;

/// <summary>
/// Resposta do resumo da dashboard para o domínio de Cadastro.
/// Retorna contagens agregadas das entidades principais.
/// </summary>
public record DashboardResumoResponse(
    long TotalObras,
    long TotalFonogramas,
    long TotalTitulares,
    long TotalAssociacoes,
    IReadOnlyList<DashboardAlerta> Alertas);

public record DashboardAlerta(string Tipo, string Mensagem);

File: services/arrecadacao-api/arrecadacao-application/src/main/java/br/com/ecad/arrecadacao/application/dto/DashboardResumoResponse.java
package br.com.ecad.arrecadacao.application.dto;

import java.math.BigDecimal;
import java.util.List;

/**
 * Resposta do resumo da dashboard para o domínio de Arrecadação.
 * Retorna métricas agregadas de licenças, verbas e pagamentos.
 */
public record DashboardResumoResponse(
        BigDecimal arrecadacaoMes,
        long totalLicencasAtivas,
        long totalLicencasSuspensas,
        BigDecimal verbaLiquidaEstimada,
        List<DashboardAlerta> alertas) {

    public record DashboardAlerta(String tipo, String mensagem) {}
}
